src/ProductConstructionService/ProductConstructionService.WorkItems/ReplicaWorkItemProcessorStateFactory.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Azure.ResourceManager.AppContainers;
+using Azure.ResourceManager.AppContainers.Models;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using ProductConstructionService.Common;
+
+namespace ProductConstructionService.WorkItems;
+public interface IReplicaWorkItemProcessorStateFactory
+{
+    Task<List<WorkItemProcessorState>> GetAllWorkItemProcessorStatesAsync();
+}
+
+public class ReplicaWorkItemProcessorStateFactory : IReplicaWorkItemProcessorStateFactory
+{
+    private readonly ContainerAppResource _containerApp;
+    private readonly IRedisCacheFactory _redisCacheFactory;
+    private readonly IServiceProvider _serviceProvider;
+
+    private List<WorkItemProcessorState>? _states = null;
+
+    public ReplicaWorkItemProcessorStateFactory(ContainerAppResource containerApp, IRedisCacheFactory redisCacheFactory, IServiceProvider serviceProvider)
+    {
+        _containerApp = containerApp;
+        _redisCacheFactory = redisCacheFactory;
+        _serviceProvider = serviceProvider;
+    }
+
+    public async Task<List<WorkItemProcessorState>> GetAllWorkItemProcessorStatesAsync()
+    {
+        if (_states != null)
```

Comment:
Why not just use `Lazy<>`?

Reply:
I didn't use Lazy because we were going to initialize it always. Using lazy feels like something that should be use when we have something big to initialize, that doesn't get initialized always

Reply:
Anyway, it's removed now

Reply:
Lazy is for something that takes time to initialize so we don't do it automatically (in ctor..) but next time we want to re-use the previous value.
It's exactly for cases like you had but also can be thread-safe so that you don't need to worry about dual initialization happening at once etc.